components/provider_ppucd/src/enum_prop_mapping.rs

```diff
@@ -0,0 +1,862 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use icu_uniset::enum_props::*;
+use std::str::FromStr;
+use tinystr::TinyStr16;
+
+//
+// Single getter function for enumerated property name:
+// Enum prop name string -> Rust enum
+//
+
+fn get_enum_property_enum(name: &str) -> Option<EnumeratedProperty> {
+    match name {
+        "bc" => Some(EnumeratedProperty::BidiClass),
+        "bpt" => Some(EnumeratedProperty::BidiPairedBracketType),
+        "ccc" => Some(EnumeratedProperty::CanonicalCombiningClass),
+        "dt" => Some(EnumeratedProperty::DecompositionType),
+        "ea" => Some(EnumeratedProperty::EastAsianWidth),
+        "gc" => Some(EnumeratedProperty::GeneralCategory),
+        "GCB" => Some(EnumeratedProperty::GraphemeClusterBreak),
+        "hst" => Some(EnumeratedProperty::HangulSyllableType),
+        "InPC" => Some(EnumeratedProperty::IndicPositionalCategory),
+        "InSC" => Some(EnumeratedProperty::IndicSyllabicCategory),
+        "jg" => Some(EnumeratedProperty::JoiningGroup),
+        "jt" => Some(EnumeratedProperty::JoiningType),
+        "lb" => Some(EnumeratedProperty::LineBreak),
+        "lccc" => Some(EnumeratedProperty::LeadCanonicalCombiningClass),
+        "NFC_QC" => Some(EnumeratedProperty::NFCQuickCheck),
+        "NFD_QC" => Some(EnumeratedProperty::NFDQuickCheck),
+        "NFKC_QC" => Some(EnumeratedProperty::NFKCQuickCheck),
+        "NFKD_QC" => Some(EnumeratedProperty::NFKDQuickCheck),
+        "nt" => Some(EnumeratedProperty::NumericType),
+        "SB" => Some(EnumeratedProperty::SentenceBreak),
+        "tccc" => Some(EnumeratedProperty::TrailCanonicalCombiningClass),
+        "vo" => Some(EnumeratedProperty::VerticalOrientation),
+        "WB" => Some(EnumeratedProperty::WordBreak),
+        _ => None,
+    }
```

Comment:
`suggestion`: I don't know if you generated it or wrote manually, but it's often more readable to put the `Option` monad outside of the match clause, for example by:

```rust
fn foo(x: &str) -> Option<Kind> {
    Some(
        match x {
            "foo" => Kind::Variant1,
            "bar" => Kind::Variant2,
            _ => return None,
        }
    )
}
```

it may seem small but on files like this it scales and the `Some` by almost all options accumulates both in size and in read cognitive load.

Reply:
Done.